lib/core/connection_factory/http.rb

```diff
@@ -5,7 +5,9 @@
 module Core
   module ConnectionFactory
     class Http
-      def self.build(url, timeout: 5, open_timeout: 5, retries: 2)
+      TIMEOUT = ENV.fetch('FRONTEND_HTTP_REQUEST_TIMEOUT').to_i
```

Comment:
First of all I understood your points and **they are valid points**.

But:
1) If the environment variable is missing, the app will not be started, and I am happy with that. And this exactly what I expect from the app itself. There are already some env vars on the site that has the same pattern. If it is missing the app will not start.

2) If someone set to "banana" or something similar *a timeout variable*, the person will be doing on purpose and not in a good faith. The value is already set on the PR on scripts here: https://github.com/moneyadviceservice/scripts/pull/918 And I am prefer to trust that nobody will change to "banana" on the timeout otherwise the person is not even know what they are doing. **Yes, I know, the code is making the assumption about trusting the people that we are working and I trust that no one will put banana or similar**.

And I prefer to don't do the *defensive program around this specific subject*. 
To be fair I don’t technically disagree with defensive program, in fact I agree with some aspects. However, defensive programming in my experience is makes the code less readable and adds a lot of unnecessary code that could be solved in different ways.

Again sorry with the long explanation, I just want to make sure that my opinions was read, heard and understood.

**Just for conclusion, I am really open that** ... **if the whole team agrees to change to be more defensive against "bananas" or similar, I will do it**. @jongilbraith & @Tr4pSt3R What are your thoughts around this?